mne_bids/path.py

```diff
@@ -405,6 +405,25 @@ def update(self, check=None, **entities):
         self._check()
         return self
 
+    def match(self):
+        """Get a list of all matching paths in the root directory.
+
+        Performs a recursive search, starting in ``.root`` (if set), based on
+        `BIDSPath.entities` object.
+
+        Returns
+        -------
+        paths : list of BIDSPath
+            The matching paths.
+        """
+        if self.root is None:
+            raise RuntimeError('Cannot match basenames if `root` '
+                               'attribute is not set. Please set the'
+                               'BIDS root directory path to `root` via '
+                               'BIDSPath.update().')
+        paths = _get_matched_bids_paths(self.root, **self.entities)
```

Comment:
why do you still need the private function. Just get rid of it, no?

Reply:
- The Main reason we discussed was that we felt it would make it easier to use this functionality in the CLI: we could use the private function directly, instead of having to first create a BIDSPath. We'll see how that goes as I continue working on #491
- I think it makes the class that uses this function easier to read as we're "offloading" the lines of code :)